tests/test_fix_flats.py

```diff
@@ -0,0 +1,373 @@
+import pytest
+import numpy as np
+from overflow.fix_flats import (
+    flat_edges,
+    label_flats,
+    away_from_higher,
+    towards_lower,
+    resolve_flats,
+    d8_masked_flow_dirs,
+)
+from overflow.constants import (
+    FLOW_DIRECTION_UNDEFINED,
+    FLOW_DIRECTION_EAST,
+    FLOW_DIRECTION_WEST,
+    FLOW_DIRECTION_SOUTH_EAST,
+    FLOW_DIRECTION_SOUTH_WEST,
+    FLOW_DIRECTION_SOUTH,
+    FLOW_DIRECTION_NORTH_EAST,
+    FLOW_DIRECTION_NORTH_WEST,
+    FLOW_DIRECTION_NORTH,
+)
+
+
+@pytest.fixture(name="dem")
+def fixture_dem():
+    """Create a dem for testing.
+
+    Returns:
+        np.ndarray: A dem with a flat that drains
+    """
+    return np.array(
+        [
+            [1, 1, 1, 1, 1, 1, 1],
+            [1, 0, 0, 0, 0, 0, 1],
+            [1, 0, 0, 0, 0, 0, 1],
+            [1, 0, 0, 0, 0, 0, 1],
+            [1, 0, 0, 0, 0, 0, 1],
+            [1, 0, 0, 0, 0, 0, 1],
+            [1, 1, -1, 1, 1, 1, 1],
+        ],
+        dtype=np.float32,
+    )
```

Comment:
Maybe instead make tests for smaller matrices, (3x3, 4x4) so it's easier to reason about

Reply:
Agreed it is a bit large, but it is taken from the example in the paper so I'm inclined to keep it as a reference we can compare with